### PR TITLE
New login page

### PR DESCRIPTION
**Pull request by Or-Geva**
- [x] All [tests](https://github.com/jfrog/jfrog-vscode-extension#building-and-testing-the-sources) passed. If this feature is not already covered by the tests, I added new tests.
- [x] I used `npm run format` for formatting the code before submitting the pull request.
-----
This pull request enhances the login view by introducing the following improvements:

1)  A new web view alongside the existing interactive login, providing a more versatile login experience.
2) It introduces a new Single Sign-On (SSO) login option, enhancing the authentication choices available to users (hidden).
3) The JFrog-CLI and Environment Variables login methods have been optimized for auto-connection.

Dependencies:
* https://github.com/jfrog/jfrog-ide-webview/pull/22
* https://github.com/jfrog/jfrog-client-js/pull/87

### CODE REVIEW

File: src/main/webview/event/tasks/login.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        // Send initial page status to the webview
+        await this.send.loadPage(this.updatedPageStatus);
+
+        // Perform login and update page status
+        const requestStatus: LoginProgressStatus = await this.doLogin();
+        await this.send.loadPage({ ...this.updatedPageStatus, status: requestStatus });
+
+        // Connect to the IDE if login is successful
+        if (requestStatus === LoginProgressStatus.Success) {
+            await this.connectIde();
```
Comment (yahavi) [inline src/main/webview/event/tasks/login.ts:48]: Question - does it block the UI? If so, can we remove the "await"?

**Review by yahavi — CHANGES_REQUESTED** [review]

File: src/main/connect/connectionUtils.ts [hunk 2]
```diff
@@ -130,7 +131,7 @@ export class ConnectionUtils {
             } else if (prompt) {
                 vscode.window.showErrorMessage((<any>error).message || error, <vscode.MessageOptions>{ modal: true });
             } else {
-                logger.logMessage((<any>error).message, 'DEBUG');
+                logger.logMessage(<any>error, 'DEBUG');
```
[the remark this answers is not in the document]
Comment (Or-Geva) [reply src/main/connect/connectionUtils.ts:134]: I was not able to gather any clear information from the `.message` property. I don't see any harm to bring all the information we have in debug mode.

File: src/main/connect/connectionUtils.ts [hunk 3]
```diff
@@ -268,7 +269,10 @@ export class ConnectionUtils {
         password: string,
         accessToken: string,
         retries?: number,
-        timeout?: number
+        timeout?: number,
+        logger?: LogManager,
+        retryOnStatusCode?: RetryOnStatusCode,
+        retryDelay?: number
```
[the remark this answers is not in the document]
Comment (Or-Geva) [reply src/main/connect/connectionUtils.ts:275]: See `getWebLoginAccessToken`

File: src/main/webview/event/tasks/login.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    private accessToken?: string;
+
+    constructor(private send: EventSender, data: ISendLoginEventData, private connectionManager: ConnectionManager, private logManager: LogManager) {
+        this.updatedPageStatus = {
+            pageType: PageType.Login,
+            url: data.url || '',
+            status: LoginProgressStatus.Verifying,
+            connectionType: data.loginConnectionType
+        };
+        this.platformUrl = data.url !== undefined ? ClientUtils.addTrailingSlashIfMissing(data.url) : '';
```
[the remark this answers is not in the document]
Comment (Or-Geva) [reply src/main/webview/event/tasks/login.ts:27]: No, that's unnecessary. This function serves as a constructor, and the appropriate protections are handled within each login flow, as demonstrated in the `doLogin` function.

File: src/main/webview/event/tasks/login.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        }
+    }
+    /**
+     * Connects to the IDE after a successful login.
+     */
+    private async connectIde() {
+        await new Promise(resolve =>
+            setTimeout(() => {
+                resolve(vscode.commands.executeCommand('jfrog.xray.connect'));
+            }, 3000)
```
[the remark this answers is not in the document]
Comment (Or-Geva) [reply src/main/webview/event/tasks/login.ts:58]: The purpose behind this approach is to provide users with a 3-second window to view the 'success' animation and read the text message.

[the remark this answers is not in the document]
Comment (Or-Geva) [reply src/main/webview/event/tasks/login.ts:58]: I will mention why we are doing this in a comment:

```js
    /**
    * Connects to the IDE after a successful login.
    * Waits for 3 seconds to let the user see the 'success' animation and read the instructions.
    */
    private async connectIde() {
        await new Promise(resolve =>
            setTimeout(() => {
                resolve(vscode.commands.executeCommand('jfrog.xray.connect'));
            }, 3000)
        );
    }
```

File: src/main/webview/event/tasks/login.ts (hunk 1 above)
Comment (Or-Geva) [reply src/main/webview/event/tasks/login.ts:48]: Removed

File: src/main/webview/event/tasks/login.ts [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+                case LoginConnectionType.Cli:
+                    status = await this.connectionManager.tryCredentialsFromJfrogCli();
+                    break;
+                case LoginConnectionType.EnvVars:
+                    status = await this.connectionManager.tryCredentialsFromEnv();
+                    break;
+            }
+            return this.toWebviewLoginStatus(status);
+        } catch (error) {
+            this.logManager.logMessage(`Failed to sign in. Error: ${JSON.stringify(error)}`, 'ERR');
```
[the remark this answers is not in the document]
Comment (Or-Geva) [reply src/main/webview/event/tasks/login.ts:92]: Following a discussion with Yahav, we have decided to keep it as it is for the time being. We will address the error related to HTML errors in the future.

**Pull request merged**